### PR TITLE
docs(scaling): Updates the partition reassignment procedure for PEM

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**
Updates the [Generating reassignment JSON files](https://strimzi.io/docs/operators/in-development/configuring.html#proc-generating-reassignment-json-files-str) procedure for scaling clusters.

The steps have been updated so that thy describe how to use PEM certs instead of PKCS #12 (ca.P12)

**NOTE** 

- Updates assume we can reference the certs and private key of the user in files.
- Not sure if there's anything about certificate chains we need to capture.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/configuring/proc-generating-reassignment-json-files.adoc [hunk 1]
```diff
@@ -127,46 +130,58 @@ Replace _<interactive_pod_name>_ with the name of the pod.
...
 security.protocol=SSL <2>
-ssl.truststore.location=/tmp/ca.p12 <3>
-ssl.truststore.password=__<truststore_password>__ <4>
-ssl.keystore.location=/tmp/user.p12 <5>
-ssl.keystore.password=__<keystore_password>__ <6>
+ssl.truststore.type=PEM <3>
+ssl.truststore.location=/tmp/ca.crt <4>
+ssl.keystore.type=PEM 
+ssl.keystore.location=/tmp/user.crt <5>
+ssl.keystore.key=/tmp/user.key  <6>
```
[the remark this answers is not in the document]
Comment (PaulRMellor) [reply documentation/modules/configuring/proc-generating-reassignment-json-files.adoc:177]: @shubham -- add key directly?

**Comment by PaulRMellor** [comment]
@ShubhamRwt and @scholzj : commit https://github.com/strimzi/strimzi-kafka-operator/pull/7568/commits/7e853713a40ed8b31dc0b56134e1c5dce090dd8f adds the user cert and key directly to the config. There's also a note on using the single-line format

**Review by scholzj — COMMENTED** [review]
Sorry, remind me again why are we doing the change from the P12 files to the PEM files? I assume it will work as documented. But:
* Inlining the files into the config is quite clumsy
* It seems a bit weird to inline the keystore but reference to the truststore in file

I think this works fine when used inside an application where this is done programatically for example through injection from environment variables. But in here it looks like we are replacing something relatively simple (use existing PKCS12 files form the secrets) with something fairly complicated.

So I wonder if we should either stick with the old way using PKCS12 files or if we should make it a tad smarter for example by using the FileConfigProvider to load the certificates which would at least make it easy to just copy paste the config file.

**Comment by PaulRMellor** [comment]
> Sorry, remind me again why are we doing the change from the P12 files to the PEM files? I assume it will work as documented. But:
> 
> * Inlining the files into the config is quite clumsy
> * It seems a bit weird to inline the keystore but reference to the truststore in file
> 
> I think this works fine when used inside an application where this is done programatically for example through injection from environment variables. But in here it looks like we are replacing something relatively simple (use existing PKCS12 files form the secrets) with something fairly complicated.
> 
> So I wonder if we should either stick with the old way using PKCS12 files or if we should make it a tad smarter for example by using the FileConfigProvider to load the certificates which would at least make it easy to just copy paste the config file.

We discussed it on a doc sync call: 

> Use PEM files in [Generating reassignment JSON files](https://strimzi.io/docs/operators/in-development/configuring.html#proc-generating-reassignment-json-files-str)? 
> Let’s change. Ask shubham for help in checking

Maybe we can show both p12 and pem approach with `FileConfigProvider`?

**Comment by ppatierno** [comment]
My 2 cents, because I helped Shubham to have the PEM working and it's really a not usable way with the command line tool.
Tbh I also think that the current PKCS12 usage is much better in this case.
Regarding mixing `ssl.truststore.location` with `ssl.keystore.certificate.chain`, while still helping Shubham I was hoping to avoid to provide the PEM content but just the file. It worked fine for the truststore but not for the keystore. After investigating the Kafka codebase it turned out that if you have a key, it forces you to use `ssl.keystore.certificate.chain` and don't allow `ssl.keystore.location`. It was a try to reduce at the minimum the number of PEMs content to put in the configuration file.

**Pull request closed without merging**